git-workflows/scripts/gather-commit-context.sh

```
@@ -0,0 +1,244 @@
+
+# Get the script directory for calling other scripts
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+# Function to get current branch
+get_current_branch() {
+  git branch --show-current 2>/dev/null || echo "HEAD"
+}
+
+# Function to check GPG signing config
+check_gpg_signing() {
+  local gpg_enabled=$(git config --get commit.gpgsign 2>/dev/null || echo "false")
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)